pipenv_setup/setup_updater.py

```diff
@@ -147,18 +147,21 @@ def format_file(file):  # type: (Path) -> None
     try:
         # noinspection PyPackageRequirements
         import black
-
+    else:
         with Popen(
             [sys.executable, "-m", "black", str(file)], stdout=PIPE, stderr=PIPE
         ) as p:
             p.communicate()
 
     except ImportError:
         # use autopep8
-        import autopep8
-
-        code = autopep8.fix_code(file.read_text())
-        file.write_text(code)
+        try:
+            import autopep8
+        else:
+            code = autopep8.fix_code(file.read_text())
+            file.write_text(code)
+        except ImportError:
+            raise RuntimeError("Either black or autopep8 must be installed.")
```

Comment:
The `else` before `except` block is a syntax error; I'd recommend the following:

```python
    except ImportError:
        # use autopep8
        try:
            # third party
            import autopep8
        except ImportError:
            raise RuntimeError("Either black or autopep8 must be installed.")
        else:
            code = autopep8.fix_code(file.read_text())
            file.write_text(code)
    else:
        with Popen(
            [sys.executable, "-m", "black", str(file)], stdout=PIPE, stderr=PIPE
        ) as p:
            p.communicate()
```